File: uploader/api/models/channel.py
class Channel():
    def __init__(self, name, unit, dataUnits=[]):
        self.name = name
        self.unit = unit
        self.dataUnits = dataUnits

    def getName(self):
        return self.name

    def addDataUnit(self, dataUnit):
        self.dataUnits.append(dataUnit)

    def replaceDataUnits(self, dataUnits):
        self.dataUnits = dataUnits

    def asDict(self):
        return {
            'name': self.name,
            'unit': self.unit,
            'floatData': [ unit.asDict() for unit in self.dataUnits ],
        }

File: backend/modules/uploader/api/models/channel.js
class Channel {
    constructor(name, unit, dataUnits = []) {
        this.name = name;
        this.unit = unit;
        this.dataUnits = dataUnits;
    }

    getName() {
        return this.name;
    }

    addDataUnit(dataUnit) {
        this.dataUnits.push(dataUnit);
    }

    replaceDataUnits(dataUnits) {
        this.dataUnits = dataUnits;
    }

    asObject() {
        return {
            name: this.name,
            unit: this.unit,
            floatData: this.dataUnits.map(unit => unit.asObject()),
        };
    }
}

module.exports = {
    Channel,
};